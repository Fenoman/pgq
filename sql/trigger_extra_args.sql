\set VERBOSITY 'terse'
set client_min_messages = 'warning';

create or replace function pgq.insert_event(queue_name text, ev_type text, ev_data text, ev_extra1 text, ev_extra2 text, ev_extra3 text, ev_extra4 text)
returns bigint as $$
begin
    raise warning 'insert_event(q=[%], t=[%], d=[%], 1=[%], 2=[%], 3=[%], 4=[%])',
        queue_name, ev_type, ev_data, ev_extra1, ev_extra2, ev_extra3, ev_extra4;
    return 1;
end;
$$ language plpgsql;

create table trigger_extra_args (nr int4 primary key, col1 text, col2 text);

create trigger extra_trig_1 after insert or update or delete on trigger_extra_args
for each row execute procedure pgq.logutriga('logutriga',
    'ev_extra1=(nr+3)::text', 'ev_extra2=col1||col2',
    'ev_extra3=$$333$$', 'ev_extra4=$$444$$', 'ev_type=$$badidea$$');
create trigger extra_trig_2 after insert or update or delete on trigger_extra_args
for each row execute procedure pgq.sqltriga('sqltriga',
    'ev_extra1=(nr+3)::text', 'ev_extra2=col1||col2',
    'ev_extra3=$$333$$', 'ev_extra4=$$444$$', 'ev_type=$$badidea$$');

-- test insert
insert into trigger_extra_args values (1, 'col1', 'col2');

-- test update
update trigger_extra_args set col1 = 'col1x', col2='col2x' where nr=1;

-- test delete
delete from trigger_extra_args where nr=1;

-- restore
drop table trigger_extra_args;
\set ECHO none
\i functions/pgq.insert_event.sql

