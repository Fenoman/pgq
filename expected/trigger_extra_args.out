\set VERBOSITY 'terse'
set client_min_messages = 'warning';
create or replace function pgq.insert_event(queue_name text, ev_type text, ev_data text, ev_extra1 text, ev_extra2 text, ev_extra3 text, ev_extra4 text)
returns bigint as $$
begin
    raise warning 'insert_event(q=[%], t=[%], d=[%], 1=[%], 2=[%], 3=[%], 4=[%])',
        queue_name, ev_type, ev_data, ev_extra1, ev_extra2, ev_extra3, ev_extra4;
    return 1;
end;
$$ language plpgsql;
create table trigger_extra_args (nr int4 primary key, col1 text, col2 text);
create trigger extra_trig_0 after insert or update or delete on trigger_extra_args
for each row execute procedure pgq.jsontriga('jsontriga',
    'ev_extra1=(nr+3)::text', 'ev_extra2=col1||col2',
    'ev_extra3=$$333$$', 'ev_extra4=$$444$$', 'ev_type=$$badidea$$');
create trigger extra_trig_1 after insert or update or delete on trigger_extra_args
for each row execute procedure pgq.logutriga('logutriga',
    'ev_extra1=(nr+3)::text', 'ev_extra2=col1||col2',
    'ev_extra3=$$333$$', 'ev_extra4=$$444$$', 'ev_type=$$badidea$$');
create trigger extra_trig_2 after insert or update or delete on trigger_extra_args
for each row execute procedure pgq.sqltriga('sqltriga',
    'ev_extra1=(nr+3)::text', 'ev_extra2=col1||col2',
    'ev_extra3=$$333$$', 'ev_extra4=$$444$$', 'ev_type=$$badidea$$');
-- test insert
insert into trigger_extra_args values (1, 'col1', 'col2');
WARNING:  insert_event(q=[jsontriga], t=[badidea], d=[{"nr":1,"col1":"col1","col2":"col2"}], 1=[4], 2=[col1col2], 3=[333], 4=[444])
WARNING:  insert_event(q=[logutriga], t=[badidea], d=[nr=1&col1=col1&col2=col2], 1=[4], 2=[col1col2], 3=[333], 4=[444])
WARNING:  insert_event(q=[sqltriga], t=[badidea], d=[(nr,col1,col2) values ('1','col1','col2')], 1=[4], 2=[col1col2], 3=[333], 4=[444])
-- test update
update trigger_extra_args set col1 = 'col1x', col2='col2x' where nr=1;
WARNING:  insert_event(q=[jsontriga], t=[badidea], d=[{"nr":1,"col1":"col1x","col2":"col2x"}], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
WARNING:  insert_event(q=[logutriga], t=[badidea], d=[nr=1&col1=col1x&col2=col2x], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
WARNING:  insert_event(q=[sqltriga], t=[badidea], d=[col1='col1x',col2='col2x' where nr='1'], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
-- test delete
delete from trigger_extra_args where nr=1;
WARNING:  insert_event(q=[jsontriga], t=[badidea], d=[{"nr":1,"col1":"col1x","col2":"col2x"}], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
WARNING:  insert_event(q=[logutriga], t=[badidea], d=[nr=1&col1=col1x&col2=col2x], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
WARNING:  insert_event(q=[sqltriga], t=[badidea], d=[nr='1'], 1=[4], 2=[col1xcol2x], 3=[333], 4=[444])
-- restore
drop table trigger_extra_args;
\set ECHO none
